tests/tests_unit/test_data_classes/test_assets.py

```diff
@@ -256,6 +257,22 @@ def test_validate_asset_hierarchy__orphans_given_ignore_false(self, exc_type):
         with pytest.raises(exc_type, match=r"Issue\(s\): 1 orphans$"):
             ahv.is_valid(on_error="raise")
 
+    @pytest.mark.parametrize("n", [1, 2, 3, 10])
+    def test_validate_asset_hierarchy__orphans_given_ignore_false__all_parent_external_id(self, n):
+        assets = [Asset(name=c, external_id=c, parent_external_id="foo") for c in string.ascii_letters[:n]]
+        ahv = AssetHierarchy(assets, ignore_orphans=False).validate(on_error="ignore")
```

Comment:
What is `ahv` can you write it out? Difficult to interpret this test now.

Reply:
Does it mean `valid_asset_hierarchy`?

Reply:
Class used to be named `AssetHierarchyValidator`. Changed to hierarchy which is short and unmistakable 😅